### PR TITLE
move constructor and assignment operator, operator<

### PR DESCRIPTION
**Pull request by karunrao97**

### CODE REVIEW

File: course3/homework-string/test.cc [hunk 1]
```diff
@@ -65,19 +66,29 @@ void string_test()
...
 	assert(plusResult == "Hex123Hex");
+
+	// move contructor and assignment operator tests
+	const char* string_c_str_pointer = string3.c_str();
+	StringType string4 = std::move(string3);
+	assert(string_c_str_pointer == string4.c_str());
+
+	StringType string5;
+	string5 = std::move(string4);
+	assert(string_c_str_pointer == string5.c_str());
```
Comment (karunrao97) [inline course3/homework-string/test.cc]: The above unit tests pass for MyString, but not for std::string. This page explains why:
http://stackoverflow.com/questions/29806718/does-stdmove-on-stdstring-garantee-that-c-str-returns-same-result

Basically, it looks like SSO (short string optimization) results in move and copy doing the same thing.

File: course3/homework-string/test.cc [hunk 2]
```diff
@@ -65,14 +66,25 @@ void string_test()
 	// operator +=
 	plusResult += string2;
 	assert(plusResult == "Hex123Hex");
+
+	// move contructor and assignment operator tests
+	StringType string4 = "Note to SSO: I am not a short string, so please don't try any fancy stuff!";
```
Comment (karunrao97) [inline course3/homework-string/test.cc:99]: Unit tests now pass for std::string as well (using clang 7). I quite enjoyed this hack, and learned a few things too :)

See previous comment here: https://github.com/maciekgajewski/c-excercises/pull/26/commits/41be1231be54003e2aacf5a946798fcf1ffc0074

**Comment by maciekgajewski** [comment]
Please, merge the latest changes in master into your branch, as the tests for operator< are missing

**Comment by karunrao97** [comment]
@maciekgajewski Maybe I'm missing something, but I tried merging the latest changes in master, and don't see any unit tests for operator< (in fact, I don't see any changes since the operator+ and operator+= tests).

**Comment by maciekgajewski** [comment]
True. I must have forgot to push it :)

**Comment by karunrao97** [comment]
homework done, passes tests. made a complete mess while trying to merge with git, though :)

File: course3/homework-string/test.cc [hunk 3]
```diff
@@ -93,14 +94,24 @@ void string_test()
 	assert(StringType("0") < StringType("1"));
 	assert(!(StringType("1") < StringType("0")));
 	assert(!(StringType("1") < StringType()));
+
+	// move contructor and assignment operator tests
+	StringType string4 = "Note to SSO: I am not a short string, so please don't try any fancy stuff!";
```
Comment (karunrao97) [inline course3/homework-string/test.cc:99]: See previous comment here: 
https://github.com/maciekgajewski/c-excercises/pull/26/files/eba1c242ef2f25bc246980490ec77a1c22c0864c#r64136535

**Pull request closed without merging**